test/functional/feature_addrman.py

```diff
@@ -54,30 +61,29 @@ def run_test(self):
         self.log.info("Check that addrman from future cannot be read")
         self.stop_node(0)
         write_addrman(peers_dat, lowest_compatible=111)
-        with self.nodes[0].assert_debug_log([
-                f'ERROR: DeserializeDB: Deserialize or I/O error - Unsupported format of addrman database: 1. It is compatible with formats >=111, but the maximum supported by this version of {self.config["environment"]["PACKAGE_NAME"]} is 3.',
-                "Recreating peers.dat",
-        ]):
-            self.start_node(0)
-        assert_equal(self.nodes[0].getnodeaddresses(), [])
+        self.nodes[0].assert_start_raises_init_error(
+            expected_msg=init_error(
+                "Unsupported format of addrman database: 1. It is compatible with "
+                "formats >=111, but the maximum supported by this version of "
+                f"{self.config['environment']['PACKAGE_NAME']} is 3.: (.+)"
```

Comment:
This is somewhat confusing or designates a corrupted file, not one from the future. How could the file be format=1 and only be compatible with formats larger than 111? If the above is changed to:

```py
write_addrman(peers_dat, format=115, lowest_compatible=111)
```

it will be a bit more realistic.

Feel free to ignore as out of the scope of this PR, given that a similar message is already in `master`.

Reply:
Instead of "fixing" the test, I think rejecting formats that are less than lowest_compatible seems like a good sanity check. Though, this would be an unrelated fix in a separate pull:

```diff
diff --git a/src/addrman.cpp b/src/addrman.cpp
index 174b3a654c..bcffe64de3 100644
--- a/src/addrman.cpp
+++ b/src/addrman.cpp
@@ -243,7 +243,7 @@ void CAddrMan::Unserialize(Stream& s_)
     uint8_t compat;
     s >> compat;
     const uint8_t lowest_compatible = compat - INCOMPATIBILITY_BASE;
-    if (lowest_compatible > FILE_FORMAT) {
+    if (format < lowest_compatible || lowest_compatible > FILE_FORMAT) {
         throw std::ios_base::failure(strprintf(
             "Unsupported format of addrman database: %u. It is compatible with formats >=%u, "
             "but the maximum supported by this version of %s is %u.",

Reply:
+1, maybe each condition deserves its own error message.